.github/workflows/maven.yml

```diff
@@ -24,10 +24,10 @@ jobs:
 
     steps:
     - uses: actions/checkout@v4
-    - name: Set up JDK 17
+    - name: Set up JDK 21
```

Comment:
How about a new `maven-java21.yml`? 

Jena requires java17 and I think we ought to keep `maven.yml` at that level.

Reply:
What about using matrix for the java versions? https://github.com/OyvindLGjesdal/jena/actions/runs/11392252975

Reply:
Good idea